activerecord/test/cases/adapters/abstract_mysql_adapter/datetime_precision_quoting_test.rb

```diff
@@ -40,10 +40,10 @@ def assert_match_quoted_microsecond_datetime(match)
 
     def stub_version(full_version_string)
       @connection.stub(:get_full_version, full_version_string) do
-        @connection.schema_cache.clear!
+        @connection.pool.pool_config.instance_variable_set(:@server_version, nil)
```

Comment:
There's a delegate on pool for some of the methods on pool config so we can do the same for server_version. I'm also fine with adding an attr_accessor to PoolConfig so we can avoid calling ivar set and get for these tests. It's already a private class.